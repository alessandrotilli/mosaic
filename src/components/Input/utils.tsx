import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
