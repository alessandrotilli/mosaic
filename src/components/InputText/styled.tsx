import { styled, TextField as MUITextField } from "@material-ui/core";

export const StyledMUITextField = styled(MUITextField)({
  width: "100%",
});
