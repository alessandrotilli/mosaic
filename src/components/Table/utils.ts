export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
