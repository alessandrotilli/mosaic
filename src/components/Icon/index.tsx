import React, { cloneElement, FC, PropsWithChildren, ReactElement, useCallback, useMemo } from "react";
import { ClassNames } from "@emotion/react";
import { keyframes, Skeleton as MUISkeleton } from "@mui/material";

import { IIcon, IIconDimensions, IRenderedIcon } from "../../types/Icon";
import { logWarn } from "../../utils/logger";
import Adornment from "../Adornment";

import { iconsCatalog } from "./utils";

export const DATA_CY_DEFAULT = "icon";

const ICON_DIMENSIONS: IIconDimensions = {
  small: 20,
  medium: 24,
  large: 35,
};

const Icon: FC<PropsWithChildren<IIcon>> = ({
  badge,
  children,
  dataCy = DATA_CY_DEFAULT,
  forwarded = {},
  loading = false,
  name,
  rotate = false,
  size = "medium",
  style: externalStyle,
  tooltip,
}) => {
  const dimensions = useMemo(() => {
    const dimension = ICON_DIMENSIONS[size];
    return { height: dimension, width: dimension };
  }, [size]);

  const props = useMemo(() => {
    let props: IRenderedIcon = {
      ...forwarded,
      "data-cy": dataCy,
      fontSize: size,
    };

    if (externalStyle) {
      props = { ...props, style: { ...externalStyle } };
    }

    if (children) {
      props = { ...props, style: { ...props.style, ...dimensions } };
    }

    return props;
  }, [children, dataCy, dimensions, externalStyle, forwarded, size]);

  const rotateKeyframes = useMemo(
    () => keyframes`
      from: {
        transform: "rotate(0deg)";
      }
      to: {
        transform: "rotate(360deg)";
      }
  `,
    []
  );

  const renderWithAnimation = useCallback(
    (element: any) => {
      return (
        <ClassNames>
          {({ css }) => {
            let animation = "none";
            if (rotate) {
              animation = css`
                ${rotateKeyframes} 2s linear infinite
              `;
            }

            let className = css`
              animation: ${animation};
            `;

            if (forwarded.className) {
              className = forwarded.className;
            }

            return (
              <Adornment badge={badge} tooltip={tooltip}>
                {cloneElement(element as ReactElement<any>, { ...props, className })}
              </Adornment>
            );
          }}
        </ClassNames>
      );
    },
    [badge, forwarded, props, rotate, rotateKeyframes, tooltip]
  );

  if (loading) {
    return <MUISkeleton variant="rectangular" style={{ ...dimensions }} />;
  }

  if (children) {
    return renderWithAnimation(children);
  }

  if (!name) {
    logWarn("Icon", "Skip rendering, both children and name are not set");
    return null;
  }

  const icon = iconsCatalog[name];
  return renderWithAnimation(icon);
};

export default Icon;
