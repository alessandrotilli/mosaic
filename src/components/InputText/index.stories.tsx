/* eslint-disable @typescript-eslint/no-explicit-any */
import { Meta, StoryObj } from "@storybook/react";
import { configure, expect, fn, userEvent, within } from "@storybook/test";

import { Icons } from "../../types/Icon";
import { logInfo } from "../../utils/logger";
import getDocsPage from "../../utils/stories";
import FormDecorator from "../../utils/stories/decorators/Form";
import LocaleDecorator, { MessageMock } from "../../utils/stories/decorators/Locale";

import { DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS, LocalizedInputText } from ".";

configure({ testIdAttribute: "data-cy" });

const COMPONENT_NAME = "InputText";
LocalizedInputText.displayName = COMPONENT_NAME;

const meta = {
  title: "Inputs/InputText",
  component: LocalizedInputText,
  decorators: [FormDecorator, LocaleDecorator],
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Text Field",
          url: "https://mui.com/components/text-fields/",
        },
        component: COMPONENT_NAME,
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          dataCyShortcut: DATA_CY_SHORTCUT,
        },
        localizableProps: LOCALIZABLE_PROPS,
      }),
      description: {
        component: "The InputText is a text field component.",
      },
    },
  },
} satisfies Meta<typeof LocalizedInputText>;

export default meta;
type Story = StoryObj<typeof meta>;

const onChangeMock = fn((value: any) => logInfo(COMPONENT_NAME, `onChange handler: value '${value}'`));

export const Primary: Story = {
  args: {
    label: "Label",
    onChange: onChangeMock,
    value: "Value",
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const inputText = canvas.getByTestId(DATA_CY_DEFAULT);
    if (!inputText) {
      return;
    }

    await userEvent.type(inputText, "Changed");
    await expect(onChangeMock).toHaveBeenCalledTimes(onChangeMock.mock.calls.length);
    await expect(onChangeMock).toHaveBeenLastCalledWith("ValueChanged");
  },
};

export const Adornment: Story = {
  args: {
    ...Primary.args,
    adornment: {
      icon: Icons.close,
    },
  },
};

export const AdornmentClickable: Story = {
  args: {
    ...Primary.args,
    adornment: {
      icon: Icons.close,
      onClick: () => {},
    },
  },
};

export const Disabled: Story = {
  args: {
    ...Primary.args,
    disabled: true,
  },
};

export const Localized: Story = {
  args: {
    ...Primary.args,
    localized: true,
    label: MessageMock.inputText,
    placeholder: MessageMock.placeholder,
  },
};

export const Multiline: Story = {
  args: {
    ...Primary.args,
    multiline: {
      rows: 3,
      rowsMax: 5,
    },
  },
};

export const Placeholder: Story = {
  args: {
    label: "Label",
    placeholder: "Enter value...",
    value: undefined,
  },
};

export const Required: Story = {
  args: {
    ...Primary.args,
    required: true,
  },
};

export const SizeSmall: Story = {
  args: {
    ...Primary.args,
    size: "small",
  },
};

export const Styled: Story = {
  args: {
    ...Primary.args,
    style: {
      color: "red",
      fontWeight: "bold",
      fontSize: "large",
      textAlign: "center",
    },
  },
};

export const VariantFilled: Story = {
  args: {
    ...Primary.args,
    variant: "filled",
  },
};

export const VariantStandard: Story = {
  args: {
    ...Primary.args,
    variant: "standard",
  },
};
