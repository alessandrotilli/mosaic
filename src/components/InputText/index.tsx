import React, { ChangeEvent, CSSProperties, FC, useCallback, useMemo } from "react";
import { TextField as MUITextField } from "@material-ui/core";

import { IInputText } from "../../types/InputText";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";
import { getAdornment } from "../Input/utils";

const BASE_STYLE: CSSProperties = { width: "100%" };

export const DATA_CY_DEFAULT = "input-text";
export const DATA_CY_SHORTCUT = "label";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [
  { name: "label", type: "string" },
  { name: "placeholder", type: "string" },
];

const InputText: FC<IInputText> = ({
  adornment: externalAdornment,
  dataCy = DATA_CY_DEFAULT,
  disabled = false,
  label,
  multiline,
  onChange,
  placeholder,
  required = false,
  shrink,
  size = "medium",
  style,
  type = "text",
  value = "",
  variant = "outlined",
}) => {
  const adornment = useMemo(() => getAdornment(externalAdornment), [externalAdornment]);

  const onChangeHandler = useCallback(
    ({ target: { value } }: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => onChange && onChange(value),
    [onChange]
  );

  return (
    <MUITextField
      disabled={disabled}
      InputLabelProps={{
        shrink,
      }}
      inputProps={{
        "data-cy": dataCy,
        style: { ...BASE_STYLE, ...style },
      }}
      InputProps={{
        endAdornment: adornment,
        readOnly: disabled,
      }}
      label={label}
      margin="normal"
      multiline={!!multiline}
      minRows={multiline?.rows}
      maxRows={multiline?.rowsMax}
      onChange={onChangeHandler}
      placeholder={placeholder}
      required={required}
      size={size}
      style={{ ...BASE_STYLE }}
      type={type}
      variant={variant}
      value={value}
    />
  );
};

export const InputTextWithProps = InputText;

export default localized(InputText, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});
