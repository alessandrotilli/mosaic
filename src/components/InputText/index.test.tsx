import renderer from "react-test-renderer";

import { Icons } from "../../types/Icon";
import { InputSize, InputType, InputVariant } from "../../types/Input";
import { IInputText } from "../../types/InputText";
import { MessageMock } from "../../utils/mocks/IntlProviderMock";
import { getTestableComponent, IPartialTestOptions, ITestOptions } from "../../utils/tests";

import InputText, { DATA_CY_DEFAULT } from ".";

const DEFAULT_TEST_OPTIONS: ITestOptions<IInputText> = {
  dataCy: DATA_CY_DEFAULT,
  domNode: "input",
  localized: true,
  props: {},
};

const getInputTextTestable = (options?: IPartialTestOptions<IInputText>) =>
  getTestableComponent(InputText, DEFAULT_TEST_OPTIONS, options);

describe("InputText test suite:", () => {
  it("default", () => {
    const { element, wrapper } = getInputTextTestable();
    expect(wrapper).toHaveLength(1);

    expect(wrapper.hasClass("MuiOutlinedInput-input"));
    expect(wrapper.prop("disabled")).toBeFalsy();
    expect(wrapper.prop("required")).toBeFalsy();
    expect(wrapper.prop("type")).toEqual("text");

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("dataCy", () => {
    const dataCy = "custom";
    const { element, wrapper } = getInputTextTestable({ dataCy, props: { dataCy } });
    expect(wrapper).toHaveLength(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  // TODO: improve this to check label
  it("localized", () => {
    const label = MessageMock.title;
    const { element } = getInputTextTestable({ dataCy: label, props: { label, localized: true } });

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  // TODO: improve this to check adornment icon
  it("adornment", () => {
    const onClick = jest.fn();
    const { element, wrapper } = getInputTextTestable({ props: { adornment: { icon: Icons.account, onClick } } });

    expect(wrapper.hasClass("MuiInputBase-inputAdornedEnd"));
    expect(wrapper.hasClass("MuiOutlinedInput-inputAdornedEnd"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("disabled", () => {
    const { element, wrapper } = getInputTextTestable({ props: { disabled: true } });
    expect(wrapper.prop("disabled")).toBeTruthy();

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("multiline", () => {
    const rows = 5;
    const rowsMax = 10;
    const { wrapper } = getInputTextTestable({
      domNode: "textarea",
      props: { multiline: { rows, rowsMax } },
    });

    expect(wrapper.prop("rows")).toEqual(rows);

    // TODO: investigate over here, element for textarea is not recognized
    // const snapshotWrapper = renderer.create(element).toJSON();
    // expect(snapshotWrapper).toMatchSnapshot();
  });

  it("onChange", () => {
    const onChange = jest.fn();
    const { element, wrapper } = getInputTextTestable({ props: { onChange } });

    wrapper.simulate("change", { target: { value: "Textual Value" } });
    expect(onChange).toHaveBeenCalledWith("Textual Value");

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("password", () => {
    const { element, wrapper } = getInputTextTestable({ props: { type: InputType.password } });
    expect(wrapper.prop("type")).toEqual("password");

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("placeholder", () => {
    const placeholder = "Placeholder";
    const { element, wrapper } = getInputTextTestable({ props: { placeholder } });
    expect(wrapper.prop("placeholder")).toEqual(placeholder);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("required", () => {
    const { element, wrapper } = getInputTextTestable({ props: { required: true } });
    expect(wrapper.prop("required")).toBeTruthy();

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("size", () => {
    const { element, wrapper } = getInputTextTestable({ props: { size: InputSize.small } });

    expect(wrapper.hasClass("MuiInputBase-inputMarginDense"));
    expect(wrapper.hasClass("MuiOutlinedInput-inputMarginDense"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("value", () => {
    const value = "Initial Text Value";
    const { element, wrapper } = getInputTextTestable({ props: { value } });
    expect(wrapper.prop("value")).toEqual(value);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("variant", () => {
    const { element, wrapper } = getInputTextTestable({ props: { variant: InputVariant.filled } });
    expect(wrapper.hasClass("MuiFilledInput-input"));

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });
});
