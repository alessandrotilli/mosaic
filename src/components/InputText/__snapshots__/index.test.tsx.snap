// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText test suite: adornment 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={false}
      id="mui-8"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        data-cy="icon-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-cy="icon-button-icon"
          data-testid="AccountCircleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: dataCy 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="custom"
      disabled={false}
      id="mui-4"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: default 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: disabled 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-readOnly MuiInputBase-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={true}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={true}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: localized 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    Title
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="locale.title"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-yjsfm1"
      >
        <span>
          Title
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: onChange 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={false}
      id="mui-12"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: password 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="password"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: placeholder 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={false}
      id="mui-16"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Placeholder"
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: required 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={true}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: size 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={false}
      id="mui-20"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: value 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-cy="input-text"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value="Initial Text Value"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputText test suite: variant 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
  style={
    {
      "width": "100%",
    }
  }
>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
      data-cy="input-text"
      disabled={false}
      id="mui-24"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      style={
        {
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
  </div>
</div>
`;
